runtime/appruntime/metrics/test_exporter.go

```diff
@@ -0,0 +1,136 @@
+package metrics
```

Comment:
I think it would be nice to move this file to a separate package so as to not pollute the metrics API with the test functionality. Move to a `metricstest` subdirectory?

Reply:
Great point, will do!